packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,208 @@
+import type { DrawerProps } from './types';
+import { Box } from '~components/Box';
+import BaseBox from '~components/Box/BaseBox';
+import { castWebType, makeMotionTime, useTheme } from '~utils';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { componentZIndices } from '~utils/componentZIndices';
+import { useDrawerStack } from '~components/Drawer/StackProvider';
+import { makeAccessible } from '~utils/makeAccessible';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { useId } from '~utils/useId';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+import { useVerifyAllowedChildren } from '~utils/useVerifyAllowedChildren';
+
+const SHOW_DRAWER = 'show-drawer';
+
+const AnimatedDrawerContainer = styled(BaseBox)<{ isFirstDrawerInStack: boolean }>(
+  ({ theme, isFirstDrawerInStack }) => {
+    return {
+      opacity: 0,
+      transform: 'translateX(0%)',
+      transition: `all
+      ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+      ${castWebType(theme.motion.easing.exit.revealing)}`,
+
+      [`&.${SHOW_DRAWER}`]: {
+        opacity: 1,
+        transform: 'translateX(-100%)',
+        transition: `all ${castWebType(makeMotionTime(theme.motion.duration.gentle))} ${castWebType(
+          theme.motion.easing.entrance.revealing,
+        )}`,
+      },
+    };
+  },
+);
+
+const DrawerOverlay = styled(FloatingOverlay)(({ theme }) => {
+  return {
+    opacity: 0,
+    transition: `opacity
+      ${makeMotionTime(theme.motion.duration.xmoderate)}
+      ${castWebType(theme.motion.easing.exit.revealing)}`,
+    backgroundColor: theme.colors.overlay.background.subtle,
+
+    [`&.${SHOW_DRAWER}`]: {
+      opacity: 1,
+      transition: `opacity ${makeMotionTime(theme.motion.duration.gentle)} ${castWebType(
+        theme.motion.easing.entrance.revealing,
+      )}`,
+    },
+  };
+});
+
+const _Drawer = ({
+  isOpen,
+  onDismiss,
+  zIndex = componentZIndices.drawer,
+  children,
+  accessibilityLabel,
+  showOverlay = true,
+  initialFocusRef,
+  testID,
+}: DrawerProps): React.ReactElement => {
+  const closeButtonRef = React.useRef<HTMLDivElement>(null);
+
+  useVerifyAllowedChildren({
+    children,
+    componentName: 'Drawer',
+    allowedComponents: [drawerComponentIds.DrawerHeader, drawerComponentIds.DrawerBody],
+  });
+
+  const { theme } = useTheme();
+  const drawerId = useId('drawer');
+  const { drawerStack, addToDrawerStack, removeFromDrawerStack } = useDrawerStack();
+
+  const { isMounted, isVisible } = usePresence(isOpen, {
+    enterTransitionDuration: theme.motion.duration.gentle,
+    exitTransitionDuration: theme.motion.duration.xmoderate,
+    initialEnter: true,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+  const stackingLevel = drawerStack.indexOf(drawerId) + 1;
+  const isFirstDrawerInStack = stackingLevel === 1 && drawerStack.length > 1;
```

Comment:
Do we need to check `drawerStack.length > 1`?

Reply:
yes. First drawer is always stackingLevel 1 but we want this check to be true only when there is another drawer on top of it.